### PR TITLE
add v9.5 transit new features

### PR DESCRIPTION
**Pull request by yueshuaing**
code changes:
- [x] 1. add different walk perception factors
- [x] 2. report out number of trimmed demand to output_summaries
- [x] 3. report out boardings at stations by access mode to output_summaries
- [x] 4. report out BART to BART transfers at 12th, 19th and MacArthur stations
- [x] 5. fix zone-to-zone fare calculation issue
- [x] 6. add different VDF for PNR mode
- [x] 7. change normalized gap and relative gap in stopping criteria
- [x] 8. remove extra parking penalty at four PNR locations (Glen Park, Lake Merritt, San Bruno, Fruitvale) because of feature NO. 6
- [x] 9. update node-specific transfer boarding time penalties at BART 19th street and Macarthur stations (change from 0.01 to 0)
- [x] 10. add transfer boarding time penalty skim
- [x] 11. code 12th, 19th and MacArthur stations as parameter instead of hard coded in the scripts.

network changes:
- [x] 1. add missing @tollbooth values (via Lasso preprocessing jupyter notebook).
- [x] 2. remove boarding and alighting prohibitions at 19th street and macarthur station (via project cards).
- [x] 3. remove boarding and alighting prohibitions on AM yellow lines (via Emme transaction file).
- [x] 4. fix bus capacity issue on PM PNR dummy route (via Lasso).

### CODE REVIEW

File: tm2py/components/network/transit/transit.py [hunk 1]
```diff
@@ -1108,9 +1113,10 @@ def run_skims(self,
                 num_processors=num_processors,
             )
 
-            walk_perception_factor = self.controller.config.transit.get("walk_perception_factor", 2)
             drive_perception_factor = self.controller.config.transit.get("drive_perception_factor", 2)
-            # divide drive and walk time by mode specific perception factor to get the actual time
+            walk_speed = 3.0
```
Comment (i-am-sijia) [inline tm2py/components/network/transit/transit.py]: Can we expose this in the model_config.toml?

File: tm2py/components/network/create_tod_scenarios.py [hunk 2]
```diff
@@ -237,18 +238,21 @@ def _create_transit_scenarios(self):
             # set fixed guideway times, and initial free flow auto link times
             # TODO: cntype_speed_map to config
             cntype_speed_map = {"CRAIL": 45.0, "HRAIL": 40.0, "LRAIL": 30.0, "FERRY": 15.0}
+            walk_speed = 3.0
+            transit_speed = 30.0
```
Comment (i-am-sijia) [inline tm2py/components/network/create_tod_scenarios.py]: Can we expose these in model_config.toml?

File: tm2py/config.py [hunk 3]
```diff
@@ -1225,7 +1226,10 @@ class TransitConfig(ConfigItem):
     output_skim_matrixname_tmpl: str = Field()
     output_transit_boardings_path: str = Field()
     output_shapefile_path: str = Field()
+    output_transit_segment_path: str = Field()
     output_station_to_station_flow_path: str = Field()
+    output_transfer_at_station_path: str = Field()
+    output_trimmed_demand: str = Field()
```
Comment (i-am-sijia) [inline tm2py/config.py]:
```suggestion
    output_trimmed_demand_report_path: str = Field()
```

File: examples/model_config.toml [hunk 4]
```diff
@@ -1532,19 +1536,19 @@
     name = "walk"
     type = "WALK"
     assign_type = "AUX_TRANSIT"
-    speed_miles_per_hour = 3.0
+    speed_miles_per_hour = "ul2*1"
```
Comment (i-am-sijia) [inline examples/model_config.toml]: Is this related to https://github.com/camsys/tm2py/blob/13eee239341e355ffd90b14148de031a42ff9332/tm2py/components/network/create_tod_scenarios.py#L286-L292

The comment on line 287 suggests ul2 is time, and you are using ul2*1 for speed here? Is it correct?

**Review by i-am-sijia — CHANGES_REQUESTED** [review]

File: tm2py/components/network/transit/transit.py (hunk 1 above)
Comment (yueshuaing) [reply tm2py/components/network/transit/transit.py]: added to model_config.toml in https://github.com/camsys/tm2py/pull/53/commits/cfce21695d62ed72e73dd31a858b23d9291cda59

File: tm2py/components/network/create_tod_scenarios.py (hunk 2 above)
Comment (yueshuaing) [reply tm2py/components/network/create_tod_scenarios.py]: added to model_config.toml in https://github.com/camsys/tm2py/pull/53/commits/cfce21695d62ed72e73dd31a858b23d9291cda59

File: tm2py/config.py (hunk 3 above)
Comment (yueshuaing) [reply tm2py/config.py]: path name updated in https://github.com/camsys/tm2py/pull/53/commits/cfce21695d62ed72e73dd31a858b23d9291cda59

File: examples/model_config.toml (hunk 4 above)
Comment (yueshuaing) [reply examples/model_config.toml]: Based on Emme mode definition , the mode speed can be a constant (speed) or a function of time (link variable). In this PR, I'm using ul2 to represent the perceived walk time.

**Review by i-am-sijia — APPROVED** [review]

**Pull request merged**